src/validation.cpp

```diff
@@ -2580,10 +2580,12 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         return error("%s: Consensus::CheckBlock: %s", __func__, FormatStateMessage(state));
 
     // Check that all non-zero coinbase outputs pay to the required destination
-    BOOST_FOREACH(const CTxOut& txout, block.vtx[0]->vout) {
-        if (chainparams.CoinbaseDestination() != CScript() && txout.scriptPubKey != chainparams.CoinbaseDestination() && !(txout.nValue.IsExplicit() && txout.nValue.GetAmount() == 0))
-            return state.DoS(100, error("ConnectBlock(): Coinbase outputs didnt match required scriptPubKey"),
-                             REJECT_INVALID, "bad-coinbase-txos");
+    if (chainparams.GetConsensus().mandatory_coinbase_destination != CScript()) {
+        BOOST_FOREACH(const CTxOut& txout, block.vtx[0]->vout) {
+            if (txout.scriptPubKey != chainparams.GetConsensus().mandatory_coinbase_destination && !(txout.nValue.IsExplicit() && txout.nValue.GetAmount() == 0))
```

Comment:
Mhmm, so if outputs are confidential or 0 it is allowed that they're sent anywhere? why?
Also shouldn't we validate that all coinbase outputs are explicit regardless of mandatory_coinbase_destination?

Reply:
By consensus they can't be confidential(the check is later in VerifyCoinbaseAnounts), and you need to allow 0-value outputs for commitments, such as segwit.